app/web/features/search/FilterDialog.tsx

```diff
@@ -212,15 +238,39 @@ export default function FilterDialog({
               />
 
               <Select
-                id="can_host_status_filter"
-                value={hostingStatusFilter}
-                onChange={(e) => {
-                  setHostingStatusFilter(e.target.value as number);
-                }}
+                id="can_host_status_filter_1"
+                variant="outlined"
                 label={t("search:form.host_filters.hosting_status_field_label")}
+                multiple={true}
+                menuItems={true}
+                value={hostingStatusFilter}
+                onChange={(e) =>
+                  setHostingStatusFilter(e.target.value as number[])
```

Comment:
why this onChange function? I did based on the MUI documentation, basically they are doing this to show as demo

![image](https://github.com/user-attachments/assets/6e447648-d0ed-4e04-98a4-665fde39b730)
![image](https://github.com/user-attachments/assets/22d759c4-7721-4196-9cb5-5739feb77f59)

Reply:
https://v4.mui.com/components/selects/#multiple-select

Reply:
a wait, of course, I receive directly an `string[]`, so I it always replace the state by the received array (even if the values don't change).  
it's how it's done in the official docs... so I guess it's fine doing it this way (the only downside I can think it's about re-updating even when the values are the same, but the dropdown doesn't allow the user to do it, so I don't think we need the complex function which searches, compares and updates the value)

Reply:
Ah okay I see it is sending an array in this case.

Please fix the types though. You should not need `as number[]` on line 248 or on line 256. That means the type is defined wrong somewhere higher up.

Reply:
Also what happens if the user wants to search for someone to meet up with and doesn't care about hosting status?